libsolidity/ast/AST.h

```diff
@@ -851,6 +853,12 @@ class CallableDeclaration: public Declaration, public VariableScope
 	ASTPointer<OverrideSpecifier> const& overrides() const { return m_overrides; }
 	std::vector<ASTPointer<VariableDeclaration>> const& returnParameters() const { return m_returnParameters->parameters(); }
 	ParameterList const& parameterList() const { return *m_parameters; }
+	SourceLocation const  prototypeLocation  () const override {
```

Comment:
The current implementation makes some assumptions. These should be documented:

```suggestion
	SourceLocation const  prototypeLocation  () const override {
		solAssert(location().contains(parameterList().location()));
```